WordPress/src/main/java/org/wordpress/android/ui/reader/actions/ReaderBlogActions.java

```diff
@@ -478,9 +484,7 @@ public static void undoBlockBlogFromReader(final BlockedBlogResult blockResult)
         if (blockResult == null) {
             return;
         }
-        if (blockResult.deletedPosts != null) {
-            ReaderPostTable.addOrUpdatePosts(null, blockResult.deletedPosts);
-        }
+       undoBlockBlogLocal(blockResult);
```

Comment:
The `undoBlockBlogFromReader` isn't invoked just when the user clicks on the "Undo" button but also if the user clicks on "block site" button and the request fails. Since we don't know the state on the server, there isn't much we can do imo. We assume that if the second request fails (undo), the first request (block site) probably failed as well.

Reply:
AFAIU we call `undoBlockBlogFromReader` only when the user clicks on the "Undo" button. This method includes server call to undo blog. 
When the user clicks on the "block site" button and the request fails, we just call `undoBlockBlogLocal` that only reverts updates to local db.
I meant to say that if the "undo" request fails inside `undoBlockBlogFromReader` we revert to original state. Here we know that undo action has failed on the server.

Having said that, I'm fine with this assumption as they happen in quick succession:
> We assume that if the second request fails (undo), the first request (block site) probably failed as well.

Reply:
Ohh, you are right. We can handle it. However, I think it's probably not worth the effort due to the assumption mentioned above. Wdyt?

Either case, good catch :)

Reply:
Let's go with this assumption 👍